### PR TITLE
add example dispatch call to template

### PR DESCRIPTION
**Pull request by rdwallis**
This adds an example RPC dispatch call to the project template.

Eventually this should be configurable so the user can select JSONDispatch instead or neither.

### CODE REVIEW

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessagePresenter.java.template [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        
+        getView().setUiHandlers(this);
+        this.dispatch = dispatch;
+        this.messageResultProvider = messageResultProvider;
+    }
+
+    @Override
+    public void sendMessage(String message) {
+        dispatch.execute(new SendMessageAction(message), new AsyncCallback<MessageResult>() {
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessagePresenter.java.template]: remove new line

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessageView.java.template [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import com.google.gwt.user.client.ui.HasText;
+import com.google.gwt.user.client.ui.HasWidgets;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.gwtplatform.mvp.client.ViewWithUiHandlers;
+
+class SendMessageView extends ViewWithUiHandlers<SendMessageUiHandlers> implements SendMessagePresenter.MyView {
+    interface Binder extends UiBinder<Widget, SendMessageView> {
+    }
+
+    @UiField HasText message;
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessageView.java.template]: We usually keep the annotation on the line before the field :
```
@UiField
HasText message;
```

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessageView.java.template [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        initWidget(uiBinder.createAndBindUi(this));
+    }
+
+    @UiHandler("sendButton")
+    void onSendButtonClick(ClickEvent event) {
+        getUiHandlers().sendMessage(message.getText());
+    }
+
+    @Override
+    public void addToSlot(Object slot, IsWidget content) {
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessageView.java.template]: Move the public method before the package private

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessagePresenter.java.template [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    interface MyView extends View, HasUiHandlers<SendMessageUiHandlers> {
+    }
+
+    @ProxyStandard
+    @NameToken(NameTokens.home)
+    interface MyProxy extends ProxyPlace<SendMessagePresenter> {
+    }
+    
+    private final DispatchAsync dispatch;
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessagePresenter.java.template]: Remove new line

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessagePresenter.java.template [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    private final Provider<MessageResultPresenter> messageResultProvider;
+    
+    static final Object MESSAGE_RESULT_SLOT = new Object();
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/SendMessagePresenter.java.template]: Move the static field before private final fields

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/messageresult/MessageResultPresenter.java.template [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    interface MyView extends View  {
+        void setMessage(String message);
+    }
+
+    @Inject
+    MessageResultPresenter(
+            EventBus eventBus,
+            MyView view) {
+        super(eventBus, view);
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/client/application/sendmessage/messageresult/MessageResultPresenter.java.template]: remove new line

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/server/message/MessageHandlerModule.java.template [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package com.example.myproject.server.message;
+
+import com.example.myproject.shared.message.SendMessageAction;
+import com.gwtplatform.dispatch.rpc.server.guice.HandlerModule;
+
+public class MessageHandlerModule extends HandlerModule {
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/server/message/MessageHandlerModule.java.template]: Remove

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/server/message/SendMessageHandler.java.template [hunk 8]
```diff
@@ -0,0 +1,27 @@
+package ${packageName}.server.message;
+
+import ${packageName}.shared.message.MessageResult;
+import ${packageName}.shared.message.SendMessageAction;
+import com.gwtplatform.dispatch.rpc.server.ExecutionContext;
+import com.gwtplatform.dispatch.rpc.server.actionhandler.AbstractActionHandler;
+import com.gwtplatform.dispatch.shared.ActionException;
+
+public class SendMessageHandler extends AbstractActionHandler<SendMessageAction, MessageResult> {
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/server/message/SendMessageHandler.java.template]: Remove

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/DefaultActionImpl.java.template [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Base abstract implementation of
+ * {@link com.gwtplatform.dispatch.shared.Action}.
+ *
+ * @param <R>
+ *            The {@link com.gwtplatform.dispatch.shared.Result} type returned.
+ */
+public abstract class DefaultActionImpl<R extends Result> implements Action<R> {
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/DefaultActionImpl.java.template]: Remove

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/message/MessageResult.java.template [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+import com.gwtplatform.dispatch.rpc.shared.Result;
+
+public class MessageResult implements Result{
+    private static final long serialVersionUID = 1L;
+    private String message;
+    
+    // GWT serialization requires a default constructor
+    @SuppressWarnings("unused")
+    private MessageResult() {
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/message/MessageResult.java.template]: REmove

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/message/SendMessageAction.java.template [hunk 11]
```diff
@@ -0,0 +1,32 @@
+package ${packageName}.shared.message;
+
+import ${packageName}.shared.DefaultActionImpl;
+
+public class SendMessageAction extends DefaultActionImpl<MessageResult>{
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/message/SendMessageAction.java.template]: Remove

File: gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/message/SendMessageAction.java.template [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+import ${packageName}.shared.DefaultActionImpl;
+
+public class SendMessageAction extends DefaultActionImpl<MessageResult>{
+    
+    private String message;
+
+    // gwt requires a default constructor for serialization.
+    @SuppressWarnings("unused")
+    private SendMessageAction() {
+
```
Comment (meriouma) [inline gwtp.plugin.core/src/main/resources/templates/project/src/main/java/${packageFolder}/shared/message/SendMessageAction.java.template]: Remove

**Comment by meriouma** [comment]
Few nits, otherwise LGTM

**Comment by rdwallis** [comment]
@meriouma thanks for the review.

This should be added behind a project option, where you can choose RPC dispatch / REST dispatch or neither.  Or might be better as part of a create Action/Result pair function.

Going to close and am unlikely to add this functionality in the near future.

**Pull request closed without merging**